TilingScoping.md

```diff
@@ -0,0 +1,152 @@
+
+As the existing API is implemented across all existing readers and writers, it may be that the cost of reworking it is too large. As such it may be best to keep the existing functions and improve upon them.
+
+The current readers largely support tiling and require little change. The main issues are with regards the writers and documentation of current support levels.
+
+How can users discover if a format supports tiling?
+We currently have no way to provide this information to users. We could improve our current handling by providing the following feedback:
+
+### **Improve documentation regarding supported formats**
+Use isFullPlane to return suitable exceptions when a format is not supported
+Implement a new function in FormatWriter to allow the user to query tiling support
+boolean supportsTiling()
+
+### **How can users find valid parameters for tiling?**
+The other major issue we currently have is that we have no way of informing the users on which parameters are suitable. Adding to this is the fact that not all writers make use of the user provided parameters. Improvements to be made here include:
+
+Make greater use of existing optimalWidth and Height,
+Improve documentation around these functions
+Fix existing methods to make full use of user provided parameters
+
+### **Pros**		
+Reduces the effort required to deliver a solution
+Keeping the existing API means we will not be making breaking changes
+
+### **Cons**
+Functions are still accessible for all writers even if not supported
+
+
+## **Option2 - New TiledWriter Interface**
+
+### **Reader Functionality - Keep existing API**
+As the existing API function for reading a tile is implemented across the board to a good standard I would propose that we keep the existing reader API. 
+
+`byte[] openTile(int no, byte[] buf, int x, int y, int w, int h)`
+`int getOptimalTileWidth()`
+`int getOptimalTileHeight()`
+ 
+### **Deprecate Existing Writer API function** 
+As the writing of tiles is only supported by a subset of our writers I would suggest deprecating the existing API function.
+
+`void saveBytes(int no, byte[] buf, int x, int y, int w, int h)`
+
+### **New TiledWriter Interface**
+For format writers which do support tiling a separate interface to handle the writing of tiles could be implemented by this subset. The functionality from the existing saveBytes would be transferred (and improved to make full use of parameters) to a new saveTile function. Additionally, I would suggest adding new functions to retrieve suitable height and width (similar to the reader API) for supported format writers.
+
+| TileWriter                                                       |
+|------------------------------------------------------------------|
+|                                                                  |
+| `+void saveTile(int no, byte[] buf, int x, int y, int w, int h)` <br/> `+int getOptimalTileHeight()` <br/> `+int getOptimalTileWidth()`|
+
```

Comment:
Additional items for consideration with a new interface which were raised in initial discussion:
- getTileHeightMultiple()
- getTileWidthMultiple()
- Ability to set the tiling size in advance
- Functions to return potential Maximum tile size
- Any new API should be built to allow for the future additions of extra dimensions not just XY
- We need to begin for the testing infrastructure and benchmarking of tiling